### PR TITLE
add editable textbox

### PR DESCRIPTION
**Pull request by esoadamo**
- Adds a textbox that can be edited by clicking on the pen on the right
- Supports theme changing
- Implements form control

Snímek z 2019-08-26 12-50-14
-------
Snímek z 2019-08-26 12-50-10
-------
Snímek z 2019-08-26 12-50-05
-------
Snímek z 2019-08-26 12-49-58

### CODE REVIEW

File: projects/is-editable-textbox/src/lib/is-editable-textbox.component.scss [hunk 1]
```diff
@@ -0,0 +1,56 @@
+$is-color-invalid: var(--is-control-border-invalid-color, red);
+$is-color-pending: var(--is-control-border-pending-color, yellow);
+$is-color-handle: var(--is-control-handle-color, #000);
+$is-control-fg-color: var(--is-control-fg-color, #000);
```
Comment (lzoubek) [inline projects/is-editable-textbox/src/lib/is-editable-textbox.component.scss]: Please import `....../scss/variables` instead

File: projects/is-editable-textbox/src/lib/is-editable-textbox.component.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<div *ngIf="edit" class="input-group input-group-text">
+  <input #textInput
+         type="text"
+         class="form-control is-input"
```
Comment (lzoubek) [inline projects/is-editable-textbox/src/lib/is-editable-textbox.component.html]: do not use 'is-input' class (I've been using it in cron-editor, but it needs to be removed as well). Import `..../mixins` and apply appropriate mixins

File: projects/is-editable-textbox/package.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "is-editable-textbox",
+  "version": "7.0.0",
```
Comment (lzoubek) [inline projects/is-editable-textbox/package.json]: Please set version to 7.1.0 .. all components who support theming are 7.1.0

File: projects/is-editable-textbox/src/lib/is-editable-textbox.component.html [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+         [autocomplete]="autocomplete"
+         [disabled]="disabled"
+         [placeholder]="placeholder"
+         [(ngModel)]="value"
```
Comment (lzoubek) [inline projects/is-editable-textbox/src/lib/is-editable-textbox.component.html]: can you get rid of [ngModel] and thus remove `FormsModule` import?

File: projects/is-editable-textbox/src/lib/is-editable-textbox.component.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    if (event) {
+      this._onChangeCallback(event.target.value);
+    }
+  }
+
+  /**
+   * Implemented as part of ControlValueAccessor.
+   */
+  writeValue(value: string): void {
+    this.value = value;
```
Comment (lzoubek) [inline projects/is-editable-textbox/src/lib/is-editable-textbox.component.ts:57]: I think you'd need to call changeDetector here. Try adding a demo code which will call `control.setValue('') within a setTimeout.

Comment (esoadamo) [reply projects/is-editable-textbox/src/lib/is-editable-textbox.component.ts:57]: You were right, it didn't work when using timeout

**Pull request merged**